lib/msal-angular/docs/0.x-1.x-upgrade-guide.md

```diff
@@ -0,0 +1,58 @@
+
+### MsalModule.forRoot now takes two arguement.
+
+Previously, MSAL Angular accepted one configuration object via `MsalModule.forRoot()`. To more closely align with `msal@1` and to provide more flexibility to MSAL Angular, this has been split into two objects, one for the core library and one for the wrapper.
+
+Steps:
+1. The first argument is the configuration object, which is the [same `Configuration` object](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-core/src/Configuration.ts) you would pass to `msal`.
+2. The second argument is a [`MsalAngularConfiguration object`](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-angular/src/msal-angular.configuration.ts), containing the values for `consentScopes`, `popUp`, `extraQueryParameters`, `protectedResourceMap`, and `unprotectedResources`.
+
+See the [updated sample](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/samples/angular6-sample-app/src/app/app.module.ts) for an example of how to pass these configuration objects.
+
+### Mitigations for AOT mode errors
+
+The new `msal` configuration object takes a function for `system.logger` and `framework.protectedResourceMap`, which does not work properly when running in `aot` mode. There are now two workarounds available:
+
+1. `protectedResourceMap` has been moved to `MsalAngularConfiguration` object, and can be passed as `[string, string[]][]` or as a `Map`. `framework.protectedResourceMap` still works, but has been deprecated. See the [updated samples](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/samples/angular6-sample-app/src/app/app.module.ts) for usage.
+2. `logger` can now be set dynamically by using `MsalService.setLogger()`. See the [updated samples](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/samples/angular6-sample-app/src/app/app.component.ts) for usage.
+
+### Other breaking changes
+
+* The `acquireToken` and `login` methods now take a single `AuthenticationParameters` object as parameters.
+* `getUser()` is now `getAccount()`.
+* Broadcast events now emit objects, instead of just strings.
+* Applications using `Redirect` methods must implement the `handleRedirectCallback` method (and have it run on every page load), which will capture the result of redirect operations. See the [Angular sample](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/samples/angular6-sample-app/src/app/app.component.ts) for an example of how to implement.
+
+## Angular 6+ and rxjs@6
+
+MSAL Angular now expects that your application is built with `@angular/core@>=6`, `@angular/core@>=6`, `rxjs@6`, and `rxjs-compat` is no longer required.
+
+Steps:
+1. Install newer versions of Angular and rxjs: `npm install @angular/core @angular/common rxjs`
+2. Uninstall `rxjs-compat`: `npm uninstall rxjs-compat`
```

Comment:
in the changelog, you mention that "`rxjs-compat` is no longer required." as though you still had the option to use it, yet this suggests that we actually recommend to not use it at all. we should probably make this idea clearer, as it isn't even clear to me which it is by reading this

Reply:
Good point. I don't believe it will cause any harm to have it installed, and I suppose it might be possible that an app needs it for compatibility with other libraries. I'll make sure its clear that it is no longer needed for MSAL Angular itself.